### ISSUE TITLE
Choice start position

### ISSUE THREAD

**Issue by Amazinite** [endless-sky/endless-sky#1519]
The ability to choose your starting planet by being given a choice at the beginning. 
So be given a choice like:

```
Where do you wish your starting location to be?
    Dirt Belt
    Paradise Worlds
    The Deep
    Core
    Pirate World
```

or even:

```
What are you?
    Human
    Hai
    Korath
```

and each choice would direct you to a different start with different start conditions and start conversations.

```
choice start
    `What are you?`
    choice
        `   Human`
            goto human

    label human
    `Where do you want to start?`
    choice
        `   The Deep`
            start deep

start deep
    date 16 11 3013
    system "Naos"
    planet "Asgard"
    set "Deep Born"
    ...

conversation deep
    `You step off the front steps of the university in your city, a new graduate ready to leave your mark on the galaxy...`
    ...
```

Would definitely help for mods that modify the start, not having to remove those plugins if you wish to have a normal start, as well as opening up new possibilities.

**Comment by KiLEdEnNis** [comment]
not only would it help mods, something like this would eventually raise ES to a whole new level, there isn't a huge team working on it but being a different species or starting somewhere else with other missions will basically make it a triple A game in my opinion

**Comment by endless-sky** [comment]
If the player decided to start as a different species, pretty much every mission in the game would need to be rewritten, So, at that point you're probably talking about a total conversion mod.

As for starting on different planets, a more general solution might just be to allow mission actions to "warp" the player to an arbitrary other location. That's something I'd already considered for other reasons (adding missions where it turns out the Quarg have some sort of instantaneous travel system between their ringworlds).

**Comment by Bladewood** [comment]
wait, what was that last part?
they have a super jump drive built into their ringworlds?

**Comment by Amazinite** [comment]
> If the player decided to start as a different species, pretty much every mission in the game would need to be rewritten, So, at that point you're probably talking about a total conversion mod.

That was the point of the little `set "Deep Born"` line in the start section; you could have certain missions that only offer if you chose a certain start position, or are restricted due to your starting position. So you wouldn't necessarily need to rewrite all the missions to accommodate for being a different species, you could just have them all require you to have a `"Human"` condition.

**Comment by endless-sky** [comment]
I've got another story line on the back burner for when the Wanderers are done - and the Wanderer plot up until when the eye opens will only cover about 1/3 of what I've planned, so that might take a while. But as a sort of "tier 2.5" mission, I'd like to have a plot involving a group of species that kicked the Quarg out of their space and took over the Quarg ringworld without really knowing all the details of how it operates. If you sides with the Quarg against them, you'd be able to make use of transporters that let you jump from one ringworld to another - including their captured ringworld, because they don't even realize it's capable of that.

Anyway, that's not exactly part of the canon yet, but I thought it'd be a pretty cool idea.

**Comment by KiLEdEnNis** [comment]
@endless-sky side idea about the ringworld warp, some part in the ringworld that acts similiar to a wormhole, but if you actually press somewhere in the map, that same "wormhole drive" can warp you to any ringworld
(in short; a wormhole-like thing that's linked to a group instead of 1 other place)

**Comment by The-Borg-Captain** [comment]
That would make for fast travel times

**Comment by MessyMix** [comment]
I know several people who have been working on different "character backgrounds" for different start locations, a little bit of which is given in the starting conversation.

**Comment by Amazinite** [comment]
There was a conversation on Steam about having difficulty settings for the game given that some people struggle to even pay off the initial loan. I brought up this issue and stated how its implementation could effectively result in having difficulty settings for the game in that where you start would change how hard it is to get off the ground. Example:

```
Choose your background.
	1. You grew up on New Boston, an uninteresting world located within the Dirt Belt region. You've dreamed of owning your own starship ever since the age of fifteen after your first time visiting the planet's spaceport and seeing massive ships parked there. After five long years of working at a textile mill and saving credits, you finally have enough funds to purchase a pilot's license from the Republic. You collect all your saved up credit chips and purchase the pilot's license from the spaceport. With freedom from this planet at your fingertips, you travel to the bank to take out a loan on a your first ship. (Hard/Vanilla start)
	2. You grew up on Midgard, a world located in the technologically advanced region known as the Deep. The Deep, semi-autonomous from the rest of the Republic, provides free higher education to all its citizens. Despite this, you've always had more of an interest in becoming a starship captain. After finishing your primary education, you research what it would take to get your own ship. To your joy, you find that the Deep provides low-interest loans to those who passed their primary education with high enough grades, and you just barely made the mark! You make your way to a bank near the shipyard, ready to start the next chapter of your life. (Normal start)
	3. You grew up on Mainsail, one of the luxurious Paradise Worlds terraformed to have the perfect climate. Only the richest in the galaxy are able to purchase real estate on such worlds, let alone simply rent a place to live. Your father is a wealthy businessman who earns his money by trading stocks at the galactic stock exchange on Martini. For your eighteenth birthday, your father surprises you by handing you a pilot's license, the keys to your family's private Flivver, and 30,000 credits. You hug your father and run to the Flivver, ready to go on a journey around the galaxy. (Easy start)
```

The current start would become the Dirt Belt start, the hardest out of the starts in this example given that it would have a loan with the highest interest rate. The Deep start has you on a planet that has the same ships in the shipyard as New Boston and then some, but the loan would probably not give you too many more options. The decreased interest of the loan would make it less burdensome to deal with though, even if the loan starts out bigger. The Paradise World start leaves you with no loan at all, a better ship, but less spare funds than the other starts. And that's just brainstormed stuff from the past half hour.

**Comment by Zitchas** [comment]
That sounds like a good set of options, although I think I prefer your alternative to the third one: Where the player simply starts off with a zero interest loan large enough to afford one of the slightly better ships. So the loan stays in the game, but the player doesn't have to focus on it (and has more flexibility in ship choices to go with it).

**Comment by Pointedstick** [comment]
I love it! That would be so cool!

**Comment by Amazinite** [comment]
I have a branch on my fork right now that drafts out how this might look.
https://github.com/endless-sky/endless-sky/compare/master...Amazinite:choice-starts

A new pilot will see `conversation "start"` that gives them a number of choices; right now the draft contains the Dirt Belt, Paradise, and Deep starts mentioned above, plus a Syndicate start that acts as an even more difficult beginning. Each of those choices, instead of having a `goto`, references that choice's start. 
```
	choice
		`	You grew up on New Boston, an uninteresting world located within the Dirt Belt region...`
			start "dirt belt"
```

Further down, there are multiple `start` blocks, each with a name similar to a mission or conversation.
```
start "dirt belt"
	date 16 11 3013
	system "Rutilicus"
	planet "New Boston"
	account
		credits 480000
		score 400
		mortgage Mortgage
			principal 480000
			interest 0.004
			term 365
	set "license: Pilot's"
	set "dirt born"
	set "human"
	conversation "dirt belt intro"
```

Each start references a conversation to start with. This is akin to the conversation that we currently get when we start the game, but instead of the conversation being titled just `intro`, it's the intro of whatever start we chose. (If you look at my branch right now, I have finished drafts for the Syndicate and Paradise starts, but not the Deep start.)
```
conversation "dirt belt intro"
	scene "scene/lobby"
	`The bank's architecture is what you would have called "futuristic" back when you dreamed that the future would be less squalid than the present...`
```

Then the game continues as expected from there, just with whatever start conditions the player chose.
Since I gave each start its own condition that allows the game to know which start this save file is part of, we can have different tutorials for each start. 

```
mission "Intro [0]"
	priority
	name "Passenger to <planet>"
	description "This old-timer captain offered to ride along with you to <destination>, and to give you some tips along the way."
	landing
	passengers 1
	source "New Boston"
	destination "New Greenland"
+	to offer
+		has "dirt born"
```

This allows us to make the tutorials of each start be catered to the area that the player starts in. My thought was that James would still always be the tutorial character, but we could easily make it so that each start has their own tutorial character; perhaps the Paradise Worlds tutorial character is a private instructor, and the Syndicate tutorial character is some shady character who helps you along but is trying to take advantage of you in some way (e.g. Star Barge start has you smuggling goods without your knowledge).
